### PR TITLE
[v2] fix escape sequence deprecation warnings

### PR DESCRIPTION
**Pull request by kdaily**
*Issue #, if available:*

*Description of changes:*

Fix `DeprecationWarning`s for invalid escape sequences.

All but one are converted to raw strings; one had a typo in the byte string.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/unit/customizations/emr/test_add_steps.py [hunk 1]
```diff
@@ -212,7 +212,7 @@ def test_custom_jar_step_with_all_fields(self):
             'Jar=s3://mybucket/mytest.jar,'
             'Args=arg1,arg2,MainClass=mymainclass,'
             'ActionOnFailure=TERMINATE_CLUSTER,'
-            'Properties=k1=v1\,k2=v2\,k3')
+            r'Properties=k1=v1\,k2=v2\,k3')
```
Comment (nateprewitt) [inline tests/unit/customizations/emr/test_add_steps.py:215]: This is a bit of a nit, but when we have a multi-line string, we may want to lean toward prefixing everything to make sure it's consistent/understood this is one raw string.

File: tests/unit/test_clidriver.py [hunk 2]
```diff
@@ -771,7 +771,7 @@ def test_user_agent(self, *args):
         machine = platform.machine()
         driver = create_clidriver()
         user_agent_extra_pattern = re.compile(
-            '^source/%s(\.[a-z_]+)?(\.[0-9]+)?$' % machine
+            r'^source/%s(\.[a-z_]+)?(\.[0-9]+)?$' % machine
```
Comment (nateprewitt) [inline tests/unit/test_clidriver.py:774]: While we're here we might as well do:
```suggestion
            fr'^source/{machine}(\.[a-z_]+)?(\.[0-9]+)?$'
```

**Comment by kdaily** [comment]
Rebased and squashed, waiting for CI to pass.

**Review by kyleknap — APPROVED** [review]
Looks good! :ship:

**Pull request merged**